### DISCUSSION TITLE
安装问题

### DISCUSSION THREAD

**Discussion by beastsam** [chenshuo/muduo#596] [General]
今天我在安装的时候报出如下错误
请问这个是我缺少了什么库么？还是有库没有链接进去？
谢谢各位指点！

[ 30%] Building CXX object examples/ace/ttcp/CMakeFiles/ttcp_asio_async.dir/ttcp_asio_async.cc.o
[ 30%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/InetAddress.cc.o
[ 30%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/Poller.cc.o
[ 31%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/poller/DefaultPoller.cc.o
/home/admin/tool/muduo/examples/ace/ttcp/ttcp_asio_async.cc: 在函数‘void doAccept(boost::asio::ip::tcp::acceptor&)’中:
/home/admin/tool/muduo/examples/ace/ttcp/ttcp_asio_async.cc:147:76: 错误：‘boost::asio::ip::tcp::acceptor’没有名为‘get_io_service’的成员
   TtcpServerConnectionPtr new_connection(new TtcpServerConnection(acceptor.get_io_service()));
                                                                            ^
[ 31%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/poller/EPollPoller.cc.o
make[2]: *** [examples/ace/ttcp/CMakeFiles/ttcp_asio_async.dir/ttcp_asio_async.cc.o] 错误 1
make[1]: *** [examples/ace/ttcp/CMakeFiles/ttcp_asio_async.dir/all] 错误 2
make[1]: *** 正在等待未完成的任务....
[ 31%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/poller/PollPoller.cc.o
[ 31%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/Socket.cc.o
[ 32%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/SocketsOps.cc.o
[ 32%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/TcpClient.cc.o
[ 32%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/TcpConnection.cc.o
[ 32%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/TcpServer.cc.o
[ 33%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/Timer.cc.o
[ 33%] Building CXX object muduo/net/CMakeFiles/muduo_net.dir/TimerQueue.cc.o
[ 33%] Linking CXX static library ../../lib/libmuduo_net.a
[ 33%] Built target muduo_net
make: *** [all] 错误 2

**Comment by chenshuo** [comment]
Please answer the questions in the template:

When file an issue muduo, please provide a [SSCCE](http://sscce.org):
Short, Self Contained, Correct (Compilable), Example.

Also specify the exact environment where the issue occurs:

## Linux distro and version? x86 or ARM? 32-bit or 64-bit?

## Branch (cpp98/cpp11/cpp17) and version of muduo?

## Version of cmake, gcc and boost? (If not from distro.)

**Comment by konigsberg** [comment]
Please refer to [https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/history.html](url).
I downgraded my Boost to 1.69.0 and it works.
